### ISSUE TITLE
Quality issue

### ISSUE THREAD

**Issue by nartes** [eBay/modanet#4]
The dataset is poorly labelled. And doesn't stand to the standards mentioned in the paper!

**Comment by 1e100** [comment]
I've noticed some annotations are severely misaligned. Take for instance image ID 450000. There's an Asian woman on the right side of the frame, yet annotations are on the left. This is not the only such misalignment, even a cursory inspection finds more instances. This will ruin training accuracy. Additionally a lot of the images referenced in the dataset don't seem to be present in Paperdoll dataset on which ModaNet is based.

Here are some stats as of January 9th 2019:
```
52254 images total
28897 good images
23357 bad images
265705 annotations total
150883 good annotations
114822 bad annotations
```
"Bad" images are the ones not found in PaperDoll at all, even among the files that can no longer be downloaded. "Bad" annotations are segmentation annotations associated with such images.

**Comment by 1e100** [comment]
Here's what I mean:

![450000](https://user-images.githubusercontent.com/38598618/50894441-1e040180-13b8-11e9-80b5-c7c113212501.png)

**Comment by 1e100** [comment]
In fact it seems to not be matching the image at all, not just misaligned.

**Comment by 1e100** [comment]
Ex:
![450000_2](https://user-images.githubusercontent.com/38598618/50894684-ada9b000-13b8-11e9-8d02-0b2ff6654c8c.png)

**Comment by 1e100** [comment]
Deleted my comments. Turns out my download of the paperdoll dataset was bad. The trick was to re-extract the URLs from Chictopia SQL rather than from the `*.mat` file in another github repo. My script is currently downloading images. I'll spot check and report back when it's done.

**Comment by mxk7721** [comment]
Good evening,

First of all, I appreciate eBay publicly releasing the ModaNet dataset.

I wanted to show some examples of errors that I ran across while exploring the dataset and training clothing detectors with them. Training a mask RCNN  detector with ResNet 50 backbone achieves similar results shown in Fig 4. (left) in all categories but boots and footwear (which show a significant drop in performance).

Though the polygon segmentation for some of the boxes are correct (left image below), it is sometimes the case that their corresponding bounding boxes share the same top left corner in their (though correct height and width).

Polygon Segmentation             |  Bounding Box
:-------------------------:|:-------------------------:
![gt_segm_0134764](https://user-images.githubusercontent.com/8944502/51928807-06d49480-23f7-11e9-8b28-066234b22966.jpg) | ![gt_bbox_0134764](https://user-images.githubusercontent.com/8944502/51928806-06d49480-23f7-11e9-8434-61bd163e98d4.jpg)

Below I am showing some errors in polygon annotations for the boots / footwear categories. In each of the following cases, the particular _image_id_ was associated with two annotations for the boots / footwear category. Each on the two labels was did not however segment out a single shoe as can be seen below. Annotation 1/2 shows all of the polygons for a particular annotation.

- Some annotations include polygons from different clothing items

Annotation 1             |  Annotation 2
:-------------------------:|:-------------------------:
![extra_clothing_left](https://user-images.githubusercontent.com/8944502/51929373-5ff0f800-23f8-11e9-8d01-9b8c184c365f.png) | ![extra_clothing_right](https://user-images.githubusercontent.com/8944502/51929374-5ff0f800-23f8-11e9-9f6a-d29dbabf5353.png)

- Some annotations, though belonging to a single item, were split into different annotations

Annotation 1             |  Annotation 2
:-------------------------:|:-------------------------:
![one_imtem_left](https://user-images.githubusercontent.com/8944502/51929890-954a1580-23f9-11e9-8582-6722db89a794.png) | ![one_item_right](https://user-images.githubusercontent.com/8944502/51929891-954a1580-23f9-11e9-9bb5-e39d95f3bdd3.png)

- Below annotations contain the polygons for parts of the items.

Annotation 1             |  Annotation 2
:-------------------------:|:-------------------------:
![weird_labels_left](https://user-images.githubusercontent.com/8944502/51929919-a72bb880-23f9-11e9-8080-91cfc550de61.png) | ![weird_labels_right](https://user-images.githubusercontent.com/8944502/51929920-a72bb880-23f9-11e9-87d5-67c6c8f640d5.png)

Annotation 1             |  Annotation 2
:-------------------------:|:-------------------------:
![boots_left](https://user-images.githubusercontent.com/8944502/51929932-abf06c80-23f9-11e9-941e-37357ecc061a.png) | ![boots_right](https://user-images.githubusercontent.com/8944502/51929933-abf06c80-23f9-11e9-878f-1d88a00d83e8.png)

**Comment by bittnt** [comment]
Hi, Regarding the polygon annotations, you should be able to generate the mask as we had in the paper by taking the group of the annotations. This would needs to have a code to parse through the annotation, but it should be easily done. There is also a way to parse through the annotations to fix the problem in getting mask for each shoe in the same image. We will release the mask as we had in the paper soon.

**Comment by bittnt** [comment]
Thanks to [the change of Japanese legislation](http://eare.eu/japan-amends-tdm-exception-copyright/), Kota has kindly released the full image dataset in paperdoll, so there would be no missing images in ModaNet: https://github.com/kyamagu/paperdoll/tree/master/data/chictopia

**Comment by Hliang1994** [comment]
@mxk7721 Hi, I meet a similary problem. I want to do a object detection. After downloading the train.json, I put bboxs on images. However, almost all footwears are wrong labeled. Could you tell me how do you deal with this situation？

**Comment by shaayaansayed** [comment]
@mxk7721 could you describe the training schedule you used to train your mask rcnn? Did you use pretrained COCO weights? Would you be willing to release the weights?

**Comment by hrsma2i** [comment]
I refined the bboxes' quality using the segmentation maps.

https://github.com/hrsma2i/modanet#Setup

**Comment by cad0p** [comment]
> I refined the bboxes' quality using the segmentation maps.
> 
> https://github.com/hrsma2i/modanet#Setup

hi, can you show us an example of a refinement you made?

**Comment by cad0p** [comment]
hi guys, I have a much better version of the annotations json wich you can download [here](https://drive.google.com/open?id=1CVWJ-QYhXV6y58gsV3XBu2gnfjfTKjcz)

it tries to merge and move the annotations of footwear and boots, by deleting double shapes present in more annotations and by trying to move the wrong bounding box to fix it. code in my repo:

https://github.com/cad0p/maskrcnn-modanet/blob/master/maskrcnn_modanet/fix_annotations.py

if you install my code you can replicate the instances_all.json

let me know! @mxk7721 @Hliang1994

**Comment by moganesyan** [comment]
@cad0p the instances_all.json object that you provided seems to have duplicate annotation IDs which are associated with different image_ids.

**Comment by cad0p** [comment]
By that you mean that I copied some annotations from one image to another
one inadvertently?

You can reconstruct the file by using the original one and applying the
script in my repo to it (you can apply it several times)
